articles/azure-functions/functions-reference-powershell.md

```diff
@@ -0,0 +1,424 @@
+---
+title: PowerShell developer reference for Azure Functions | Microsoft Docs
+description: Understand how to develop functions by using PowerShell.
+services: functions
+documentationcenter: na
+author: tyleonha
+manager: slee
+keywords: azure functions, functions, event processing, webhooks, dynamic compute, serverless architecture
+
+ms.assetid: 45dedd78-3ff9-411f-bb4b-16d29a11384c
+ms.service: azure-functions
+ms.devlang: powershell
+ms.topic: reference
+ms.date: 11/5/2018
+ms.author: tyleonha
+
+---
+# Azure Functions PowerShell developer guide
+
+This guide contains information about the intricacies of writing Azure Functions with PowerShell.
+
+A PowerShell function is represented as a PowerShell script that executes when triggered ([triggers are configured in function.json](functions-triggers-bindings.md)). The powershell script takes in parameters that match the names of all the input bindings. In addition to those inputs, a parameter is available to you called `TriggerMetadata` that contains additional information on the trigger that started the function.
+
+This article assumes that you have already read the [Azure Functions developer reference](functions-reference.md). You should also complete the Functions quickstart to create your first function, using [Visual Studio Code](functions-create-first-function-vs-code.md) or [via the Azure Functions Core Tools](functions-create-first-azure-function.md).
+
+## Folder structure
+
+The required folder structure for a JavaScript project looks like the following. This default can be changed. For more information, see the [scriptFile](#using-scriptfile) section below.
+
+```
+PSFunctionApp
+ | - MyFirstFunction
+ | | - run.ps1
+ | | - function.json
+ | - MySecondFunction
+ | | - run.ps1
+ | | - function.json
+ | - Modules
+ | | - myFirstHelperModule.psm1
```

Comment:
Shouldn't the individual modules be in separate child folders and with psd1 (as best practice although not required)?

Reply:
fixed